### ISSUE TITLE
cast number to string?

### ISSUE THREAD

**Issue by braco** [litixsoft/lx-valid#7]
Currently, {cast:true} will convert a string to a number. Is there any reason not to do the opposite?

year: 1972

would become

year: "1972"

but this currently produces "must be of string type"

**Comment by 4kochi** [comment]
Well, it's more common to cast a string to a number. You often get strings as an input and it's reasonable to try to convert that string into a number if you are expecting a number. But the other way round is more tricky. Not technical. But more contextual. For example if you have a zipcode or phone number. I guess it would be surprising it it's automatically converted. Maybe i would store it as number and my schema is wrong. And you also have to decide which types should be converted. Because anything could be casted to a string.

But you can actually use the convert function.

``` js
var data = {
        zipcode: 12345
    },
    schema = {
        properties: {
            zipcode: {
                type: 'string',
                format: 'integer'
            }
        }
    },
    convertFn = function (format, value) {
        if (format === 'integer') {
            return value.toString();
        }

        return value;
    },
    res = validate(data, schema, {convert: convertFn});

// birthdate was converted
console.log(typeof data.zipcode === 'string'); // true
```

**Comment by TedSoper** [comment]
Snippet doesn't work. In my data set using the same format as you the only items that get sent through the convertFn are valid values.

ie: {zipcode: 12345  } never hits convertFn but {zipcode: "12345"} does.

Seems to be the opposite of what we would need.

**Comment by 4kochi** [comment]
You are right, the snippet doesn't work. Actually the convert function only triggers if the type of the property is `string`. One would have to change this behavior so that the convert function is triggered regardless of the current type. Feel free to send a pull request if you are interested in that. We had no use case for this yet.
